lezioni/riepargomgendata.php

```diff
@@ -232,17 +247,49 @@
                             <td width=35%>Attivit&agrave;</td>";
 
                     while ($reclez = mysqli_fetch_array($rislez)) {
-                        print "<tr><td>" . $docassoclist[$reclez['iddocente']] . "</td><td>" . $matassoclist[$reclez['idmateria']] . "</td><td>" . $reclez['argomenti'] . "&nbsp;</td><td>" . $reclez['attivita'] . "&nbsp;</td></tr>";
+                        stampa_lez(
+                            $con,
+                            $docassoclist,
+                            $reclez['idlezione'], 
+                            $matassoclist[$reclez['idmateria']],
+                            $reclez['argomenti'],
+                            $reclez['attivita']
+                        );
                     }
 
                     print "</table>";
                 }
             }
         }
-        // VISUALIZZARE ARGOMENTI SOSTEGNO
     }
 }
 
 mysqli_close($con);
 
 stampa_piede("");
+
+function stampa_lez($con, $docassoclist, $idlezione, $materia, $argomenti, $attvità)
+{
+    print "<tr>";
+    //selezioniamo tutti i docenti che hanno firmato quella lezione
+    $lezquery = "SELECT `iddocente` FROM `tbl_firme` WHERE `idlezione` = $idlezione;";
+    $risdocs = eseguiQuery($con, $lezquery);
+    if(mysqli_num_rows($risdocs) == 0){
+        print "<td>Nessuna firma</td>";
+    }else{
+        //stampa tutti i docenti corrispondenti alla lezione
+        print "<td>";
+        $f = true;
+        while($docente = mysqli_fetch_array($risdocs)){
+            //stampa la virgola solo se ci sono più docenti
+            if(!$f)
+                print ", ";
+            $f = false;
+            print $docassoclist[$docente['iddocente']];
```

Comment:
Questo anche è una stringa? Già che ci siamo chiedo che imparo LAMPSchool :D

Grazie ciau!

Reply:
È il nome del docente preso dalla matrice di associazione, che risalendo all'origine dei dati li prende all'inserimento dell'anagrafica fatta dal superadmin `adminlamp`.
Viene sanificata in input quindi è safe.

Reply:
Ho appena controllato, anche le stringhe delle lezioni sono safe e sanificate in input quindi faccio revert.

Reply:
OK grazie!

In generale però ogni singola stringa stampata a video e che non deve contenere HTML facciamola passare in quella funzione. Non tanto per noi, ma per i contributori occasionali che non hanno la vostra esperienza e rischierebbero di prendere questa parte di codice e copio-incollarla in altre facendo gravi danni.

Reply:
Perfetto, per questa teniamola così, siccome potrebbe causare problemi soprattutto su campi come quelli che contengono tanti caratteri speciali. 

Propongo la creazione di delle **guidelines alla contribuzione** con tutte le best practices del caso.